vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthResetPasswordEntryFragment.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.onboarding.ftueauth
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.lifecycle.lifecycleScope
+import dagger.hilt.android.AndroidEntryPoint
+import im.vector.app.core.extensions.associateContentStateWith
+import im.vector.app.core.extensions.content
+import im.vector.app.core.extensions.editText
+import im.vector.app.core.extensions.hidePassword
+import im.vector.app.core.extensions.isEmail
+import im.vector.app.core.extensions.setOnImeDoneListener
+import im.vector.app.databinding.FragmentFtueResetPasswordInputBinding
+import im.vector.app.features.onboarding.OnboardingAction
+import im.vector.app.features.onboarding.OnboardingViewState
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import reactivecircus.flowbinding.android.widget.textChanges
+
+@AndroidEntryPoint
+class FtueAuthResetPasswordEntryFragment : AbstractFtueAuthFragment<FragmentFtueResetPasswordInputBinding>() {
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentFtueResetPasswordInputBinding {
+        return FragmentFtueResetPasswordInputBinding.inflate(inflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+    }
+
+    private fun setupViews() {
+        views.newPasswordInput.associateContentStateWith(button = views.newPasswordSubmit)
+        views.newPasswordInput.setOnImeDoneListener { resetPassword() }
+        views.newPasswordSubmit.debouncedClicks { resetPassword() }
+
+        views.newPasswordInput.editText().textChanges()
+                .onEach {
+                    views.newPasswordInput.error = null
+                    views.newPasswordSubmit.isEnabled = it.isEmail()
```

Comment:
`it.isEmail()` looks wrong here. It's the field to enter the new password, no?

Also the notice suggest that the password should have 8 chars min. But some homeserver may require other requirements. This is just to let you know that this part should be made easy to be customized for forks.

Reply:
argh this is definitely wrong, great catch! the reason why the button still works correctly is because https://github.com/vector-im/element-android/pull/6263/files#diff-c29312b48a988ddf4691b69ec9126155953a737ea5d6d5eade6a8db7bc322797R52 is doing the empty content check

Reply:
yes, thanks, I was wondering why it was working :)